core/client/fs/src/main/java/alluxio/client/block/stream/LocalFileDataReader.java

```diff
@@ -149,6 +151,10 @@ public Factory(FileSystemContext context, WorkerNetAddress address, long blockId
         OpenLocalBlockResponse response = mStream.receive(mDataTimeoutMs);
         Preconditions.checkState(response.hasPath());
         mPath = response.getPath();
+        if (!Files.exists(Paths.get(mPath))) {
```

Comment:
Can you move the mPath exist check outside this try catch block.
This try block is try to catch anything throw when creating the GrpcBlockingStream or when sending messages, so that mBlockWorker can be close.

mBlockWorker =
try {
        mStream =
        mPath =
      } catch (Exception e) {
      mBlockWorker.close();
      throw e;
}

if (!Files.exists(Paths.get(mPath))) {
          throw new NotFoundException(String.format(
                  "LocalFileDataReader can not find the path:%s", mPath));
   BlockWorker.close();
  mStream.close();
)

Reply:
Yes, thanks!

Reply:
@LuQQiu I have modified the code. PTAL